Sources/SwiftSyntaxMacros/FunctionParameterUtils.swift

```diff
@@ -1,6 +1,87 @@
 import SwiftSyntax
 
+extension TokenSyntax {
+  var isAnyOperator: Bool {
+    switch tokenKind {
+    case .binaryOperator, .prefixOperator, .postfixOperator:
+      return true
+
+    default:
+      return false
+    }
+  }
+}
+
 extension FunctionParameterSyntax {
+  /// Determine based on context whether this parameter has an argument
+  /// label by default.
+  var parameterHasArgumentLabelByDefault: Bool {
+    // Parameters are nested within a FunctionParameterListSyntax, within
+    //   - a ParameterClauseSyntax, within
+    //     - a FunctionSignatureSyntax, within one of a FunctionDeclSyntax,
+    //       an InitializerDeclSyntax, or a MacroDeclSyntax.Signature, OR
+    //     - a SubscriptDeclSyntax
+    guard let paramList = parent?.as(FunctionParameterListSyntax.self),
+          let paramClause = paramList.parent?.as(ParameterClauseSyntax.self) else {
+      return false
+    }
+
+    // Within a FunctionSignatureSyntax...
+    if let funcSignature = paramClause.parent?.as(FunctionSignatureSyntax.self) {
+      // Within a function declaration, which has argument labels by default
+      // if it's not an operator function.
+      if let funcDecl = funcSignature.parent?.as(FunctionDeclSyntax.self) {
+        return !funcDecl.identifier.isAnyOperator
+      }
+
+      // Within an initializer, which has argument labels by default.
+      if funcSignature.parent?.as(InitializerDeclSyntax.self) != nil {
+        return true
+      }
+
+      // Within a macro declaration, which has argument labels by default.
+      if funcSignature.parent?.as(MacroDeclSyntax.self) != nil {
+        return true
+      }
+
+      return false;
```

Comment:
Which cases are covered here? Shouldn’t you be explicitly covering all cases? And if so, should we `assert` or `precondition` here?

---

Also, looks like you’ve been writing some C++ again 😆 

```suggestion
      return false
```